hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -584,7 +584,10 @@ def _imports_for_strategy(strategy):
     # If we have a lazy from_type strategy, because unwrapping it gives us an
     # error or invalid syntax, import that type and we're done.
     if isinstance(strategy, LazyStrategy):
-        if strategy.function is st.from_type:
+        if (
+            strategy.function is st.from_type
+            or strategy.function.__name__ is st.from_regex.__name__
```

Comment:
Maybe `strategy.function in (st.from_type, st.from_regex)`?  

We'll also want to handle more than just the first argument, in case of e.g. `st.from_regex("pattern", flags=re.ASCII)`.  If handling flags turns out to be a huge pain I'm happy to just open an issue about that for later, but it'd be good to investigate now in case we can support that with only minor changes.

Reply:
This sadly does not work, they get different memory addresses somehow, so it needs to be done via `.__name__`.

Probably iterating on `._LazyStrategy__kwargs` and `_LazyStrategy__args` is the best bet.

Reply:
Also I just realized the `from_regex` does not support flags itself, they need to be included in the pattern after the last `/` or in the `re.compile` call